### PR TITLE
created get all users info route for admins

### PR DESCRIPTION
**Pull request by rymaju**
## Why

To create a admin only route to read all user summaries (a subset of their interesting traits that an admin may care about) to be displayed on the frontend.

Route can be accessed at: `/api/v1/protected/user/user-directory`

https://github.com/Code-4-Community/lucys-love-bus-frontend/pull/129

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java [hunk 1]
```diff
@@ -125,6 +127,28 @@ public UserInformation getPersonalUserInformation(int userId, JWTData userData)
...
+                CONTACTS.EMAIL,
+                CONTACTS.USER_ID,
+                USERS.PRIVILEGE_LEVEL,
+                CONTACTS.PHONE_NUMBER,
+                CONTACTS.PROFILE_PICTURE,
+                USERS.PHOTO_RELEASE)
+            .from(USERS)
+            .join(CONTACTS)
+            .on(USERS.ID.eq(CONTACTS.USER_ID))
+            .where(CONTACTS.IS_MAIN_CONTACT.isTrue())
```
Comment (liammoyn) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java:148]: Make sure you specify in the route documentation that this is only returning the main contact for each user account

**Review by liammoyn — APPROVED** [review]
Make sure to update the API spec!

File: api/src/main/java/com/codeforcommunity/dto/protected_user/GetAllUserInfoResponse.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.codeforcommunity.dto.protected_user;
+
+import com.codeforcommunity.dto.protected_user.components.UserSummary;
+import java.util.List;
+
+/** Represents the action of getting all user summaries for the user directory */
+public class GetAllUserInfoResponse {
+
+  private List<UserSummary> users;
```
Comment (florisdobber) [inline api/src/main/java/com/codeforcommunity/dto/protected_user/GetAllUserInfoResponse.java:9]: Generally fields are made final in response classes

**Review by florisdobber — APPROVED** [review]
Looks good to me!

**Pull request merged**